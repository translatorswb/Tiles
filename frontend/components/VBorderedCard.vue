<template>
  <v-card
    ripple
    tile
    hover
    nuxt
    class="bordered-card primary--text"
    :class="$vuetify.rtl ? 'bordered-right' : 'bordered-left'"
    @click="navigateTo"
  >
    <slot />
  </v-card>
</template>

<script>
export default {
  props: {
    to: {
      type: String,
      required: false
    }
  },
  methods: {
    navigateTo() {
      if (this.to) {
        this.$router.push(this.to);
      } else {
        this.$emit("click");
      }
    }
  }
};
</script>

<style scoped>
.bordered-card {
  padding: 16px;
}

.bordered-left {
  border-left: 8px solid #e8991c !important;
  padding-left: 16px;
}

.bordered-right {
  border-right: 8px solid #e8991c !important;
  padding-right: 16px;
}
</style>
