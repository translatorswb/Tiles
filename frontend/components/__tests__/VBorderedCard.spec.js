import Vuetify from "vuetify";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import VBorderedCard from "../VBorderedCard.vue";

const mountFunction = options => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();
  return shallowMount(VBorderedCard, {
    localVue,
    vuetify,
    ...options,
  })
}

describe("VBorderedCard", () => {
  test("renders the slot with text", () => {
    const wrapper = mountFunction({
      slots: {
        default: "Card content"
      }
    });
    expect(wrapper.text()).toContain("Card content");
  })

  test("renders the slot with template string", () => {
    const wrapper = mountFunction({
      slots: {
        default: "<div class='bordered-card-content'>Card Content</div>"
      }
    });
    expect(wrapper.find(".bordered-card-content").exists()).toBe(true);
  })

  test("renders the slot with another component", () => {
    const Foo = {
      name: 'foo-component',
      template: '<p>Foo</p>'
    }

    const wrapper = mountFunction({
      slots: {
        default: Foo
      }
    });
    expect(wrapper.find(Foo).is(Foo)).toBe(true);
  })

  test("navigates to 'to' when clicked if 'to' exists", () => {
    const mocks = {
      $router: {
        push: jest.fn()
      }
    };
    const wrapper = mountFunction({
      propsData: {
        to: "/"
      },
      mocks
    });
    wrapper.trigger("click");
    expect(mocks.$router.push).toHaveBeenCalledWith("/");
  });

  test("emits 'click' event when clicked if 'to' doesn't exist", () => {
    const wrapper = mountFunction();
    wrapper.trigger("click");
    expect(wrapper.emitted("click")).toHaveLength(1);
  });

  test("renders correctly", () => {
    const wrapper = mountFunction();
    expect(wrapper.element).toMatchSnapshot();
  })
});