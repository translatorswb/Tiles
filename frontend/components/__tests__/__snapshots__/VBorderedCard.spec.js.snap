// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBorderedCard renders correctly 1`] = `
<v-card
  class="bordered-card primary--text bordered-left"
  hover=""
  nuxt=""
  ripple=""
  tile=""
/>
`;
